subprojects/language-jvm/src/main/java/org/gradle/api/tasks/compile/AbstractCompile.java

```diff
@@ -82,9 +108,12 @@ public void setDestinationDir(File destinationDir) {
      * @param destinationDir The destination directory. Must not be null.
      *
      * @since 4.0
+     * @deprecated set the value of {@link #getDestinationDirectory()} instead
      */
+    @Deprecated
     public void setDestinationDir(Provider<File> destinationDir) {
-        this.destinationDir.set(destinationDir);
+        DeprecationLogger.nagUserOfReplacedMethod("setDestinationDir", "getDestinationDirectory");
```

Comment:
Sanity check this for me, but I think this should be written out like `AbstractCompile.setDestinationDir(Provider<File>)` and `AbstractCompile.getDestinationDirectory().set()`

Reply:
Right. Thanks. (This also moves to the follow up PR for deprecation.)